PukingPredator/Assets/Scripts/UI/Note.cs

```diff
@@ -0,0 +1,125 @@
+using System.Collections;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class Note : MonoBehaviour
+{
+    /// <summary>
+    /// The rate that the image fades in and out.
+    /// </summary>
+    [SerializeField]
+    private float fadeSpeed = 4f;
+
+    /// <summary>
+    /// If the player was nearby the last time a check was performed.
+    /// </summary>
+    private bool isPlayerNearby = false;
+
+    /// <summary>
+    /// To store the original alpha of the image.
+    /// </summary>
+    private float originalAlpha;
+
+    /// <summary>
+    /// The image shown by the note.
+    /// </summary>
+    [SerializeField]
+    private Sprite sprite;
+
+    /// <summary>
+    /// A reference to the image this note is for.
+    /// </summary>
+    private Image UIImage;
+
+
+
+    void Start()
+    {
+        CreateImageComponent();
+    }
+
+    void OnTriggerEnter(Collider other)
+    {
+        if (isPlayerNearby) { return; }
+
+        if (other.gameObject.CompareTag(GameTag.player))
+        {
+            isPlayerNearby = true;
+            StopAllCoroutines(); // Stop any ongoing fading out
+            StartCoroutine(FadeImage(originalAlpha)); // Fade in to alpha = 1 * originalAlpha

```

Comment:
from my understanding, alpha is the original(max) alpha we want this image to achieve. Could we add a [SerializeField] to this to change it in the editor maybe?

Reply:
there's not really much point, the images are intended to be fully opaque anyways. it frankly could have originalAlpha swapped to just 1 and have no real consequences in our use case